File: fastmq-common/src/main/java/com/song/fastmq/common/concurrent/SimpleThreadFactory.java
package com.song.fastmq.common.concurrent;

import com.google.common.util.concurrent.ThreadFactoryBuilder;
import java.util.concurrent.ThreadFactory;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by song on 2017/11/5.
 */
public class SimpleThreadFactory implements ThreadFactory {

    private static final Logger logger = LoggerFactory.getLogger(SimpleThreadFactory.class);

    private ThreadFactory threadFactory;

    public SimpleThreadFactory(String name) {
        ThreadFactoryBuilder builder = new ThreadFactoryBuilder();
        builder
            .setNameFormat(name)
            .setUncaughtExceptionHandler((t, e) -> logger.error("Uncaught exception", e));
        this.threadFactory = builder.build();
    }

    @Override public Thread newThread(Runnable r) {
        return threadFactory.newThread(r);
    }
}

File: fastmq-common/src/main/java/com/song/fastmq/common/concurrent/SimpleThreadFactory.kt
package com.song.fastmq.common.concurrent

import com.google.common.util.concurrent.ThreadFactoryBuilder
import org.slf4j.LoggerFactory
import java.util.concurrent.ThreadFactory

/**
 * Created by song on 2017/11/5.
 */
class SimpleThreadFactory(name: String) : ThreadFactory {

    private val threadFactory: ThreadFactory

    init {
        val builder = ThreadFactoryBuilder()
        builder
            .setNameFormat(name)
            .setUncaughtExceptionHandler { t, e -> logger.error("Uncaught exception of thread_" + t.toString(), e) }
        this.threadFactory = builder.build()
    }

    override fun newThread(r: Runnable): Thread {
        return threadFactory.newThread(r)
    }

    companion object {

        private val logger = LoggerFactory.getLogger(SimpleThreadFactory::class.java)
    }
}
